tests/header.rs

```diff
@@ -0,0 +1,64 @@
+mod fixtures;
+
+use assert_cmd::prelude::*;
+use assert_fs::fixture::TempDir;
+use fixtures::{port, tmpdir, Error};
+use rstest::rstest;
+use std::process::{Command, Stdio};
+use std::thread::sleep;
+use std::time::Duration;
+
+#[rstest]
+fn custom_header_set(tmpdir: TempDir, port: u16) -> Result<(), Error> {
```

Comment:
I think you should be able to use rstest's parametrization feature to write these tests more succinctly by parametrizing over a list of headers.